### PR TITLE
feat(linter): `eslint-plugin-vitest/require-local-test-context-for-concurrent-snapshots`

### PR DESCRIPTION
**Pull request by eryue0220**
see support [eslint-plugin-vitest/require-local-test-context-for-concurrent-snapshots](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/require-local-test-context-for-concurrent-snapshots.md)

### CODE REVIEW

File: crates/oxc_linter/src/rules/vitest/require_local_test_context_for_concurrent_snapshots.rs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    /// test.concurrent('myLogic', (context) => {
+    ///     context.expect(true).toMatchSnapshot();
+    /// })
+    /// ```
+    RequireLocalTestContextForConcurrentSnapshots,
+    style,
+);
+
+impl Rule for RequireLocalTestContextForConcurrentSnapshots {
+    fn run_once(&self, ctx: &LintContext) {
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/vitest/require_local_test_context_for_concurrent_snapshots.rs:59]: Looks like we don't need to collect function parameters? We can iterate all `expect` call expression and combine you `check` method  

https://github.com/veritem/eslint-plugin-vitest/blob/8b11357b8690c276b869a04140f5238679bab5de/src/rules/require-local-test-context-for-concurrent-snapshots.ts#L33

**Comment by mysteryven** [comment]
Could you split a PR that doesn't contain `require-local-test-context-for-concurrent-snapshots`? Let's merge the bug fixes first, maybe in #4797 ?

**Comment by eryue0220** [comment]
> Could you split a PR that doesn't contain `require-local-test-context-for-concurrent-snapshots`? Let's merge the bug fixes first, maybe in #4797 ?

Done.

**Comment by mysteryven** [comment]
Close because of duplicated: #4951.

Thanks for your work on this rule.

**Pull request closed without merging**